hedera-node/src/test/java/com/hedera/services/txns/token/TokenBurnTransitionLogicTest.java

```diff
@@ -96,6 +108,30 @@ void followsHappyPath() {
 		verify(store).persistTokenRelationship(treasuryRel);
 	}
 
+	@Test
+	void followsHappyPathForUnique(){
+		// setup:
+		treasuryRel = new TokenRelationship(token, treasury);
+
+		givenValidUniqueTxnCtx();
+		given(accessor.getTxn()).willReturn(tokenBurnTxn);
+		given(txnCtx.accessor()).willReturn(accessor);
+		given(store.loadToken(id)).willReturn(token);
+		given(token.getTreasury()).willReturn(treasury);
+		given(store.loadTokenRelationship(token, treasury)).willReturn(treasuryRel);
+		given(token.getType()).willReturn(TokenType.NON_FUNGIBLE_UNIQUE);
+		// when:
+		subject.doStateTransition();
+
+		// then:
+		verify(token).getType();
+		verify(token).burn(any(), any(), any());
```

Comment:
We can verify the exact arguments as-well no only `any`. If there is a bug and we start to pass something else, this test would not catch it